CIP-onchain-cert-metadata/README.md

```diff
@@ -0,0 +1,485 @@
+
+This hash is calculated by taking the entire metadata tree object, ordering the keys in the object alphanumerically, and then hashing the resulting JSON string using the **blake2b-256** hashing algorithm. The hash is encoded as a hex string.
+
+`schemaVersion`: A versioning number for the json schema of the on-chain metadata. This current description shall be refered as `schemaVersion: 1`.
+
+`metadata`: An array of links to the dApp's metadata. The metadata is a JSON object that contains the dApp's metadata in accordance with [CIP 26](https://cips.cardano.org/cips/cip26/)
+
+Values for all fields shall be shorter than 64 characters to be able to be included on-chain.
+
+### Metadata Label
+
+When submitting the transaction metadata pick the following value for `transaction_metadatum_label`:
+
+- `1304`: DApp Certification
+
+### Off-chain Metadata Format
+
+The Dapp Certification certificate is complemented by off-chain metadata that can be fetched from off-chain metadata servers.
+
+### Off-chain Metadata Properties
+
+`subject`, a mandatory string, Identifier of the claim subject (i.e dApp). A UTF-8 encoded string.
+
+`schemaVersion`, a mandatory string, used as a versioning number for the Json schema of the on-chain metadata. This current description shall be refered as `schemaVersion: 1`.
+
+`certificateIssuer`, a mandatory object used to describe the certification issuer, that contains the following fields:
+
+- `name`, a mandatory string, the name of the certification issuer
+- `logo`, a string, a link to the logo of the certification issuer. The logo could be self-hosted in order to allow updates on the logo design.
+- `social`, an object used to list all the different social contacts of the Certificate Issuer:
+  - `twitter`, a string, the twitter handle of the Certificate Issuer
+  - `github`, a string, the github handle of the Certificate Issuer
+  - `contact`, a mandatory string, the contact email of the Certificate Issuer
+  - `website`, a mandatory string, a link to the Certificate Issuer's website
+  - `discord`, a string, a link to the Certificate Issuer's Discord server
+
+`report`, an object that contains:
+
+- `reportURLs`, an array of URLs, is a field where each link points to the same actual certification report for anyone to read. This ensures transparency in the findings, what was and was not considered in the certification process..
+- `reportHash`, a string, is a field that is the blake2b-256 hash of the report pointed by the links in `reportURLs`.
+
+`summary`, a string, that contains the summary of the report from the certification issuer.
+
+`disclaimer`, a string, that contains the legal disclaimer from the certification issuer.
+
+`script` an array of objects that represents the whole DApp's on-chain script. Each object is comprised of:
+
+- `fullScriptHash`, a string, is the prefix and script hash or script hash+staking key
```

Comment:
I would say fullScriptHash is an artefact of history, just put scriptHash, which in fact will a full script hash. I suggest to keep contractAddress as optional, as you can derive it anyway, it is just bech32 format.